utils/handleToasts.ts

```diff
@@ -0,0 +1,52 @@
+import { toast } from 'react-toastify';
+
+function generateToastID() {
+  return checkIsIdOccupied(randomId());
+}
+
+function randomId() {
+  return Math.floor(Math.random() * 100000000).toString();
+}
+
+function checkIsIdOccupied(toastId: string) {
+  if (document.getElementById(toastId.toString()) !== null) {
+    return checkIsIdOccupied(randomId());
+  }
+  return toastId;
+}
+
+export function handleErrorToast(errorText: string) {
+  try {
+    const generatedToastID = generateToastID();
+    if (!document) {
+      return;
+    }
```

Comment:
Onko tämä tarpeellinen tsekkaus?

Puhuttiin silloin serverin puolen rendauksesta, eikö se voi tapahtua tätä funktiota kutsuessa? Silloinhan `document` ei pitäisi olla olemassa.

Jos tämä tsekkaus säilytetään, vaihtaisin paikkaa rivin 20 `generatedToastID`:n kanssa, jotta turhaan ei generoi ID:tä, jos sitä ei voidakaan käyttää. Käytännössä lienee turha muokkaus

Reply:
Tsekkaus ei ole tarpeellinen, koska `handleErrorToast` funktiota ei koskaan kutsuta serverillä.

Reply:
Varmistan vielä että ymmärsin oikein. Eli jos `handleErrorToast` kutsuttaisiinkin backendissä, silloin tarkistus on tarpeellinen?

Toinen kysymys vielä joka ei tähän varsinaisesti liity. Mutta `useEffect`-funktiota `.tsx`-päätteisissä tiedostoissa ei ajeta serverillä koskaan, jonka takia esimerkiksi seuraavia tsekkauksia ei tarvitse tehdä:

```ts
useEffect(() => {
   if(window){
   // code   
}
  }, []);
```
tai

```ts
useEffect(() => {
   if(document){
   // code   
}
  }, []);
```

Reply:
> Varmistan vielä että ymmärsin oikein. Eli jos `handleErrorToast` kutsuttaisiinkin backendissä, silloin tarkistus on tarpeellinen?

Kyllä, juurikin näin.


> Toinen kysymys vielä joka ei tähän varsinaisesti liity. Mutta `useEffect`-funktiota `.tsx`-päätteisissä tiedostoissa ei ajeta serverillä koskaan, jonka takia esimerkiksi seuraavia tsekkauksia ei tarvitse tehdä:
> 
> ```ts
> useEffect(() => {
>    if(window){
>    // code   
> }
>   }, []);
> ```
> 
> tai
> 
> ```ts
> useEffect(() => {
>    if(document){
>    // code   
> }
>   }, []);
> ```
> 
> `window` ja `document` ovat vain `client`-puolen funktioita / classeja / objekteja / tai millä nimellä kutsutaankaan


Jep `window` ja `document` ovat objekteja jotka on määritetty cain client-puolella. Mutta koska `useEffect` kuten myös kaikki `onClick` handlerit ajetaan vain client-puolella niissä tarkistusta ei tarvitse tehdä.